hack/lib/util.sh

```diff
@@ -48,6 +48,36 @@ kube::util::wait_for_url() {
   return 1
 }
 
+# Example:  kube::util::trap_add 'echo "in trap DEBUG"' DEBUG
+# See: http://stackoverflow.com/questions/3338030/multiple-bash-traps-for-the-same-signal
+kube::util::trap_add() {
+  local trap_add_cmd
+  trap_add_cmd=$1
+  shift
+
+  for trap_add_name in "$@"; do
+    local existing_cmd
+    local new_cmd
+
+    # Grab the currently defined trap commands for this trap
+    existing_cmd=`trap -p "${trap_add_name}" |  awk -F"'" '{print $2}'`
+
+    if [[ -z "${existing_cmd}" ]]; then
+      new_cmd="${trap_add_cmd}"
+    else
+      new_cmd="${existing_cmd};${trap_add_cmd}"
+    fi
+
+    # Assign the test
+    trap "${new_cmd}" "${trap_add_name}"
+  done
+}
+
+# Opposite of kube::util::ensure-temp-dir()
+kube::util::cleanup-temp-dir() {
+  rm -rf "${KUBE_TEMP}" || true
```

Comment:
> It would if you called kube::util::clenup-temp-dir() without having called kube::util::ensure-temp-dir() Doing so feels like a coding problem that should cause it to fail. In fact, the only caller is the trap from ensure-temp-dir so I think we are safe, no?

You've gone to the effort of `|| true` to prevent failures, I just expected this to go one way or the other ("you must have preconditions correct" or "cleanup will never fail").  I don't feel strongly, I just got bit by a similar bash function in openshift.